arrows/core/detected_object_set_output_kw18.cxx

```diff
@@ -85,21 +97,64 @@ detected_object_set_output_kw18()
 detected_object_set_output_kw18::
 ~detected_object_set_output_kw18()
 {
+  if( d->m_tot_writer && d->m_tot_writer->is_open() )
+  {
+    d->m_tot_writer->close();
+  }
 }
 
 
 // ------------------------------------------------------------------
 void
 detected_object_set_output_kw18::
-set_configuration(vital::config_block_sptr config)
-{ }
+set_configuration( vital::config_block_sptr config )
+{
+  d->m_write_tot = config->get_value<bool>( "write_tot" , d->m_write_tot );
+
+  d->m_tot_field1_ids = config->get_value<std::string>( "tot_field1_ids" , d->m_tot_field1_ids );
+  d->m_tot_field2_ids = config->get_value<std::string>( "tot_field2_ids" , d->m_tot_field2_ids );
+
+  vital::tokenize( d->m_tot_field1_ids, d->m_parsed_tot_ids1, ",;", true );
+  vital::tokenize( d->m_tot_field2_ids, d->m_parsed_tot_ids2, ",;", true );
+}
+
+
+// ------------------------------------------------------------------
+vital::config_block_sptr
+detected_object_set_output_kw18::
+get_configuration() const
+{
+  // get base config from base class
+  kwiver::vital::config_block_sptr config = algorithm::get_configuration();
+
+  // Class parameters
+  config->set_value( "write_tot", d->m_write_tot,
+                     "Write a file in the vpView TOT format alongside "
+                     "the computed tracks." );
+  config->set_value( "tot_field1_ids", d->m_tot_field1_ids,
+                     "Comma seperated list of ids used for TOT field 1." );
+  config->set_value( "tot_field2_ids", d->m_tot_field2_ids,
+                     "Comma seperated list of ids used for TOT field 2." );
+
+  return config;
+}
 
 
 // ------------------------------------------------------------------
 bool
 detected_object_set_output_kw18::
-check_configuration(vital::config_block_sptr config) const
+check_configuration( vital::config_block_sptr config ) const
 {
```

Comment:
The ordering isn't clear between get_configuration(), set_configuration(), and check_configuration().
In some cases check_config() is called first to validate the xxx:type specification for the generic algorithm handling. A nice log message is generated if `type` is not specified of can't be found. All available types are displayed.

This area of configuring an algorithm needs better documentation and examples. The current code does it many different ways so you cant depend on set_configuration() being called first.